File: Go/1288. Remove Covered Intervals.go
package main

import "sort"

//goland:noinspection GoUnusedFunction
func removeCoveredIntervals(intervals [][]int) int {
	length := len(intervals)

	if length == 1 {
		return length
	}

	sort.Slice(intervals, func(i, j int) bool {
		if intervals[i][0] == intervals[j][0] {
			return intervals[i][1] > intervals[j][1]
		}

		return intervals[i][0] < intervals[j][0]
	})

	count := 1
	prev := intervals[0][1]

	for i := 1; i < length; i++ {
		current := intervals[i][1]
		if current > prev {
			count++
			prev = current
		}
	}

	return count
}

File: Python/1288. Remove Covered Intervals.py
# noinspection PyMethodMayBeStatic, PyPep8Naming
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        length = len(intervals)
        if length == 1:
            return length

        intervals.sort(key=lambda x: (x[0], -x[1]))

        count = 1
        prev = intervals[0][1]

        for i in range(1, length):
            current = intervals[i][1]
            if current > prev:
                count += 1
                prev = current

        return count
